Xamarin.Forms.Core/ResourceDictionary.cs

```diff
@@ -68,7 +71,11 @@ public void SetAndLoadSource(Uri value, string resourcePath, Assembly assembly,
 			if (type != null)
 				_mergedInstance = s_instances.GetValue(type, (key) => (ResourceDictionary)Activator.CreateInstance(key));
 			else
-				_mergedInstance = DependencyService.Get<IResourcesLoader>().CreateFromResource<ResourceDictionary>(resourcePath, assembly, lineInfo);
+			{
+				_assembly = assembly;
+				_path = System.IO.Path.GetDirectoryName(resourcePath);
```

Comment:
Looking at this, "path" is not entirely correct since it is the root - maybe resourceRoot or directoryPath - something to indicate that is is not the path (which may be a file) but a directory.